### PR TITLE
docs: replace types in `Logger`

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
See #6310.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Log/Handlers/ErrorlogHandler.php [hunk 1]
```diff
@@ -62,11 +62,15 @@ public function __construct(array $config = [])
...
-     * @param string $message
+     * @param string                  $level
+     * @param array|int|object|string $message
      */
     public function handle($level, $message): bool
     {
+        if (! is_string($message)) {
+            $message = print_r($message, true);
+        }
+
```
[the remark this answers is not in the document]
Comment (ddevsr) [reply system/Log/Handlers/ErrorlogHandler.php:73]: Okay, we can accept `array|int|object` addition in function `handle`?

[the remark this answers is not in the document]
Comment (kenjis) [reply system/Log/Handlers/ErrorlogHandler.php:73]: Yes, `$message` type is `string|\Stringable`.
https://github.com/php-fig/log/blob/fe5ea303b0887d5caefd3d431c3e61ad47037001/src/LoggerInterface.php#L30

[the remark this answers is not in the document]
Comment (kenjis) [reply system/Log/Handlers/ErrorlogHandler.php:73]: I disagree, changing to a PSR-3 compatible logger may break apps.

[the remark this answers is not in the document]
Comment (ddevsr) [reply system/Log/Handlers/ErrorlogHandler.php:73]: @kenjis I found this when `git cherry-pick --continue`
```
------ ---------------------------------------------------------------------------------------------------------------------
  Line   system\CLI\Console.php
 ------ ---------------------------------------------------------------------------------------------------------------------
         Ignored error pattern #^Strict comparison using === between array<mixed, mixed> and array{} will always evaluate to
         false\.$# in path D:\Project\laragon\www\ci4\system\CLI\Console.php was not matched in reported errors.
  71     Strict comparison using === between array and array{} will always evaluate to false.
 ------ ---------------------------------------------------------------------------------------------------------------------


 [ERROR] Found 2 errors

Fix the phpstan error(s) before commit.
```

[the remark this answers is not in the document]
Comment (kenjis) [reply system/Log/Handlers/ErrorlogHandler.php:73]: I've fixed the error.
See #6627

[the remark this answers is not in the document]
Comment (ddevsr) [reply system/Log/Handlers/ErrorlogHandler.php:73]: image

In `Psr\Log\LoggerInterface` still with `string` only ([v1.1.4](https://github.com/php-fig/log/compare/1.1.4...2.0.0)), I think we set now `string` only. `\Stringable` still under develop

[the remark this answers is not in the document]
Comment (kenjis) [reply system/Log/Handlers/ErrorlogHandler.php:73]: `Stringable` can be used in PHP 8.0 or later.

File: system/Log/Handlers/ErrorlogHandler.php [hunk 2]
```diff
@@ -62,11 +62,15 @@ public function __construct(array $config = [])
...
-     * @param string $level
-     * @param string $message
+     * @param string                  $level
+     * @param array|int|object|string $message
      */
     public function handle($level, $message): bool
     {
+        if (! is_string($message)) {
+            $message = print_r($message, true);
+        }
```
Comment (kenjis) [inline system/Log/Handlers/ErrorlogHandler.php:72]: This change is not acceptable.
https://github.com/codeigniter4/CodeIgniter4/blob/15fa3f13990523e8d9a3e98a3610889719a88dac/system/Log/Logger.php#L23

**Review by kenjis — CHANGES_REQUESTED** [review]
I think $message should be strring or stringable.

**Comment by ddevsr** [comment]
> I think $message should be strring or stringable.

in `handler` function too?

File: system/Log/Handlers/ErrorlogHandler.php (hunk 2 above)
Comment (ddevsr) [reply system/Log/Handlers/ErrorlogHandler.php:72]: Still in `4.3` branch if remove this?

**Comment by kenjis** [comment]
> in handler function too?

Yes. `handler()` gets `$message` from `log_message()`.

File: system/Log/Handlers/ErrorlogHandler.php (hunk 2 above)
Comment (ddevsr) [reply system/Log/Handlers/ErrorlogHandler.php:72]: Ohh mistake, 1 hour ago phpstan release `1.8.8` and in this PR using `1.8.7` @kenjis

**Pull request closed without merging**